karaf-features-gen/src/main/groovy/com/github/lburgazzoli/gradle/plugin/KarafFeaturesGenTask.groovy

```diff
@@ -158,39 +167,40 @@ class KarafFeaturesGenTask extends DefaultTask {
      *
      * @param dep The dependency to check
      * @param extension The karafFeatures extension
-     * @param resolvedArtifactMap The map of GAV->ResolvedArtifact
      *
      * @return {@code true} to indicate that the dependency should be wrapped; {@code false} indicates it should not.
      */
-    static boolean shouldWrap(
+    boolean shouldWrap(
             ResolvedComponentResult dep,
-            KarafFeaturesGenTaskExtension extension,
+            Project selectedProject,
             Map<ModuleVersionIdentifier,ResolvedArtifact> resolvedArtifactMap) {
-        return matchesPattern( dep, extension.wraps) || !isOsgi( dep, resolvedArtifactMap )
-    }
+        if ( matchesPattern( dep, extension.wraps ) ) {
+            return true;
+        }
+
+        if ( matchesPattern( dep, "${selectedProject.group}/${selectedProject.name}/${selectedProject.version}") ) {
```

Comment:
The artifact for the project needs to be handled differently in terms of accessing its jar.  That is what this part is about.